src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala

```diff
@@ -132,12 +166,22 @@ package object servlets extends DebugEnhancedLogging {
       matchesEitherOf(archiveExtRegexp, archiveContentTypeRegexp)
     }
 
-    def getArchiveInputStream: Try[resource.ManagedResource[ArchiveInputStream]] = Try {
+    private def getArchiveInputStream: Try[resource.ManagedResource[ArchiveInputStream]] = Try {
       val charSet = fileItem.charset.getOrElse("UTF8")
       if (matchesEitherOf(s".+[.]$tarExtRegexp", tarContentTypeRegexp))
-        managed(new TarArchiveInputStream(fileItem.getInputStream, charSet))
+        managed(new TarArchiveInputStream(fileItem.getInputStream, 10240, 512, charSet, true))
       else managed(new ZipArchiveInputStream(fileItem.getInputStream, charSet, true, true))
     }
+
+    /**
+     * @param targetDir    target for the extracted files
+     * @param draftDeposit info for error messages
+     * @param path         location relative to uploadRoot of draftDeposit (info for error messages)
+     * @return
+     */
+    def unpackPlainEntriesTo(targetDir: File, draftDeposit: => DepositDir, path: => Path): Try[Unit] = {
+      getArchiveInputStream.flatMap(_.apply(_.unpackPlainEntriesTo(targetDir, draftDeposit, path, fileItem.name)))
```

Comment:
a for-comprehension would make this somewhat more readable:

```suggestion
      for {
        managedArchiveIS <- getArchiveInputStream
        _ <- managedArchiveIS.acquireAndGet(_.unpackPlainEntriesTo(targetDir, draftDeposit, path, fileItem.name))
      } yield ()
```